config/pipeline.yaml

```diff
@@ -341,6 +342,13 @@ jobs:
   kbuild-gcc-12-arm64:
     <<: *kbuild-gcc-12-arm64-job
 
+  # Default config and build only job
+  kbuild-gcc-12-arm64-build-only:
+    <<: *kbuild-gcc-12-arm64-job
+    rules:
+      tree:
```

Comment:
This makes sense overall, with #718 in mind it would be nice to have an anchor for "build-only" rules, for example:

```yaml
  build-only-trees: &build-only-trees-rules
    rules:
      tree:
      - 'arnd'
```

And then this job definition would end up as:
```yaml
  kbuild-gcc-12-arm64-build-only:
    <<: *kbuild-gcc-12-arm64-job
    rules: *build-only-trees-rules
```

Reply:
I'd created *-build-only job for some architectures. Some architectures have different name jobs, but they are build only. I'd reused those jobs. To create a uniformity, I'll create build-only-trees-rules and *-build-only job separately.

Reply:
Now, the question is: do we actually want to have a build-only job for all architectures, and do we want to execute all those jobs for all those trees? I mean, I'd expect the `amlogic` tree to be irrelevant to all architectures except `arm*`.

Reply:
There could be a division that some trees should only be tested on their intended architectures. But it would be hard to get it right. From legacy what I've understood that we run tests for all architectures minimally on these trees without much thought to actual intended architecture.

I guess at this point, we should think from just a migration point of view and port the configurations same to same from legacy to new KernelCI.